eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/client/EclairRpcClient.scala

```diff
@@ -663,22 +663,28 @@ class EclairRpcClient(val instance: EclairInstance)(
 
     val runnable = new Runnable() {
 
-      private val attempt = new AtomicInteger(1)
+      private val attempts = new AtomicInteger(1)
 
       override def run(): Unit = {
-        val resultsF = getSentInfo(paymentId)
-        resultsF.recover { case e: Throwable => logger.error("Cannot check payment status", e) }
-        for {
-          results <- resultsF
-        } yield {
-          results.find(_.status != PaymentStatus.PENDING) match {
-            case Some(result) =>
+        if (attempts.incrementAndGet() > maxAttempts) {
```

Comment:
This gets rid of a compiler warning, adds the `paymentId`/`interval` to the log message for failing the future. This also updates the scaladoc to talk about the completion scenarios for the future we are returning  

```scala
  /**
    * Pings eclair to see if a invoice has been paid
    * If the invoice has been paid or the payment has failed, we publish a
    * [[org.bitcoins.eclair.rpc.json.PaymentResult PaymentResult]]
    * event to the [[akka.actor.ActorSystem ActorSystem]]'s
    * [[akka.event.EventStream ActorSystem.eventStream]]
    *
    * We also return a Future[PaymentResult] that is completed when one of three things is true
    * 1. The payment has succeeded
    * 2. The payment has failed
    * 3. We have attempted to query the eclair more than maxAttempts, and the payment is still pending
    */
  override def monitorSentPayment(
      paymentId: PaymentId,
      interval: FiniteDuration,
      maxAttempts: Int): Future[PaymentResult] = {
    val p: Promise[PaymentResult] = Promise[PaymentResult]()

    val runnable = new Runnable() {

      private val attempts = new AtomicInteger(0)

      override def run(): Unit = {
        if (attempts.incrementAndGet() > maxAttempts) {
          // too many tries to get info about a payment
          // either Eclair is down or the payment is still in PENDING state for some reason
          // complete the promise with an exception so the runnable will be canceled
          p.failure(
            new RuntimeException(
              s"EclairApi.monitorSentPayment() too many attempts: ${attempts
                .get()} for paymentId=${paymentId} for interval=${interval}"))
        } else {
          val resultsF = getSentInfo(paymentId)
          resultsF.recover {
            case e: Throwable => logger.error("Cannot check payment status", e)
          }
          val _ = for {
            results <- resultsF
          } yield {
            results.foreach { result =>
              result.status match {
                case PaymentStatus.PENDING =>
                //do nothing, while we wait for eclair to attempt to process
                case PaymentStatus.SUCCEEDED | PaymentStatus.FAILED =>
                  // invoice has been succeeded or has failed, let's publish to event stream
                  // so subscribers to the event stream can see that a payment
                  // was received or failed
                  // complete the promise so the runnable will be canceled
                  p.success(result)
              }
            }
          }
        }
      }
    }

    val cancellable = system.scheduler.schedule(interval, interval, runnable)

    val f = p.future

    f.onComplete(_ => cancellable.cancel())
    f.foreach(system.eventStream.publish)

    f
  }
```